### ISSUE TITLE
npm run lint: Maximum call stack size exceeded on Travis

### ISSUE THREAD

**Issue by ryanbrainard** [rackt/react-redux#199]
The Travis build is currently [failing](https://travis-ci.org/rackt/react-redux/builds/92339184) on `npm run lint`:

```
$ npm run lint
> react-redux@4.0.0 lint /home/travis/build/rackt/react-redux
> eslint src test
/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:70
        if (node == null) {
                 ^
RangeError: Maximum call stack size exceeded
    at keys (native)
    at Referencer.Visitor.visitChildren (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:78:24)
    at Referencer.Visitor.visit (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:112:14)
    at Referencer.Visitor.visitChildren (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:93:26)
    at Referencer.Visitor.visit (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:112:14)
    at Referencer.Visitor.visitChildren (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:88:38)
    at Referencer.Visitor.visit (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:112:14)
    at Referencer.Visitor.visitChildren (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:93:26)
    at Referencer.Visitor.visit (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:112:14)
    at Referencer.Visitor.visitChildren (/home/travis/build/rackt/react-redux/node_modules/eslint/node_modules/escope/node_modules/esrecurse/esrecurse.js:88:38)
```

This is failing on code that previously passed on Travis (i.e. the build above is a no-op branch from master, which was green [6 days ago](https://travis-ci.org/rackt/react-redux/builds/91091135)), so there's something external happening. What I have tried so far:
  - Running locally (Mac OS 10.10.5, Node 5.0.0, Node 3.3.9). Passes.
  - Locking down the `eslint` (and related plugins) versions down. Passes locally/Fails on Travis.
  - Upgrading `eslint` (and related plugins) to the latest. Passes locally/Fails on Travis.

Perhaps this is a Mac vs Linux problem or a iojs problem with eslint?

**Comment by inakianduaga** [comment]
Now that Travis supports docker you could run the Travis build inside a docker container (such as ubuntu:trusty) and then you would have a complete lock on the environment (both OS & node version,  at least for testing linux distros).

The Travis yml would look something like this

```yml
sudo: required
dist: trusty
before_install:
- docker pull node/5
script:
- docker run --rm -i --net=none -v $PWD:/app node/5 apt-get install npm; npm install; npm run lint;
- whatever else you want to run
```

If you want to test a bunch of nodejs versions you can just add multiple `docker run` lines with the different nodejs image versions

**Comment by ellbee** [comment]
Fixed by #204
